File: langs/csharp/src/VMx.Notifications/Notification.cs
namespace VMx.Notifications;

/// <summary>
/// Immutable notification value. Two <see cref="Notification"/> instances with
/// identical <see cref="Type"/> and <see cref="Message"/> are still different
/// instances (identity-distinct per spec/16-notifications.md).
/// </summary>
public sealed class Notification
{
    /// <summary>Creates a new notification.</summary>
    public Notification(NotificationType type, string message)
    {
        Type = type;
        Message = message;
    }

    /// <summary>Notification classification.</summary>
    public NotificationType Type { get; }

    /// <summary>Human-readable message.</summary>
    public string Message { get; }
}

File: langs/python/src/vmx/notifications/notification.py
"""Notification value class. See spec/16-notifications.md."""

from __future__ import annotations

from vmx.notifications.notification_type import NotificationType


class Notification:
    """Immutable notification value (identity-distinct)."""

    __slots__ = ("_message", "_type")

    def __init__(self, type: NotificationType, message: str) -> None:
        self._type = type
        self._message = message

    @property
    def type(self) -> NotificationType:
        return self._type

    @property
    def message(self) -> str:
        return self._message
